src/rgw/rgw_op.cc

```diff
@@ -4427,10 +4428,9 @@ void RGWPutMetadataAccount::execute(optional_yield y)
 
   /* We are passing here the current (old) user info to allow the function
    * optimize-out some operations. */
-  op_ret = s->user->store_info(this, y, RGWUserCtl::PutParams()
-			     .set_old_info(&s->user->get_info())
-			     .set_objv_tracker(&acct_op_tracker)
-			     .set_attrs(&attrs));
+  s->user->set_attrs(attrs);
+  op_ret = s->user->store_by_id(this, y, false, &s->user->get_info());
```

Comment:
From what @cbodley  described, I think acct_op_tracker needs to be passed here either as argument or by setting user->objv_tracker to acct_op_tracker, unless read_attrs at line:4350 reads version directly into user->objv_tracker.

Reply:
You're half right.  The read_attrs() needs to read into user->objv_tracker, but line 4415 has already set acct_op_tracker to user->objv_tracker.  I'll fix it.